src/VsExtension/VsEvents/OnBuildPackageRestorer.cs

```diff
@@ -33,6 +34,9 @@ internal sealed class OnBuildPackageRestorer
         private readonly DTE _dte;
         private bool _outputOptOutMessage;
 
+        private IReadOnlyDictionary<string, BuildIntegratedProjectCacheEntry> _buildIntegratedCache
```

Comment:
the cache is swapped out each time, it gets the current closure, compares, and replaces it for the next time